Cl 221/error phone email field sign up in

brentguf opened the pull request:
Now the phone/email field has error messaging/validation when it's possible to sign up/in with a phone number. The code is obvious and simple, so not a lot to add. There are some comments in the code.

I'm still going to try to add unit tests to this on a different branch before releasing.

## Checklist

- [x] Added entry to changelog
<details>
<summary>More info</summary>
Add a concise line to the 'Next release' section of the changelog (docs/README.md) so people other than developers can understand what has changed where. E.g. 'Added an error message to the project name field of the project edit form (Admin > Projects > Edit)'.
</details>

- [x] WCAG 2.1 AA proof
<details>
<summary>More info</summary>
For front-end devs only. Is your work conforming with the WCAG 2.1 AA rules? If you need more info, read the [a11y page](https://www.notion.so/citizenlab/a11y-7568f83d42ab4895ac133b89d358997b) on our Notion.
</details>

- [ ] Tests
<details>
<summary>More info</summary>

### Unit tests

Did you add relevant unit tests?

### E2E tests

Sometimes it can be more efficient to update E2E tests after CI has run them. If you know which ones to update, go ahead! E2E template cl2-back:

```bash
docker compose run --rm web bin/rails cl2_back:create_tenant[localhost,e2etests_template]
```

</details>

- [x] Prepared branch for code review
<details>
<summary>More info</summary>
Reviewed code to reduce unnecessary back and forth (removal of console.log, comments, ...)? Added comments to clarify code, emphasize what to pay attention to, etc.?
</details>

## Links

- [citizenlab-ee PR](**put URL here** or remove)
- [Specs](**put URL here** or remove)
- [Epic Deployment](**put URL here** or remove)

## How urgent is a code review?

Still this week or by Monday next week would be great.

adessy committed:
IN-696 Add data-source relationships  to serialized views

adessy committed:
IN-696 Define a set of allowed origin types for data sources

adessy committed:
IN-696 Allow the creation of a view with multiple data sources via the API

adessy committed:
IN-696 Adapt Views::CreateService tests for multi-project views

adessy committed:
IN-696 Raise error when trying to create a view without data source

adessy committed:
IN-696 Remove support for view scope from the API endpoints

adessy committed:
IN-696 Implement view-policy rules based on data sources

Implement view-policy rules based on data sources instead of the view scope.

IvaKop committed:
Add project checkboxes

IvaKop committed:
Add animation on collapse/expand

IvaKop committed:
Add folder checkbox logic

IvaKop committed:
Fix borders and selection logic

IvaKop committed:
Clean up selection logic

Commit:
Translations updated by CI (extract-intl)

IvaKop committed:
FIx tests

IvaKop committed:
Add label

IvaKop committed:
Fix TopBar test

IvaKop committed:
Fix view type

IvaKop committed:
Fix type for IInsightsViewObject

IvaKop committed:
[IN-701] Fix typo and add useMemo to folder data

IvaKop committed:
[IN-701] Fix button alignment

adessy committed:
IN-696 Add category_assignments association to views

adessy committed:
IN-696 Mark all inputs from all origins as processed

Mark all inputs from all origins as processed when creating a new insights view.

adessy committed:
IN-696 Fix view policy for `create` action

adessy committed:
IN-696 Adapt view acceptance tests for multi-project insights

adessy committed:
IN-696 Remove unauthorized requests from API documentation

adessy committed:
IN-696 Use existing helper functions

adessy committed:
IN-696 Speed up acceptance tests for views

adessy committed:
IN-696 Remove reference to `view.scope.ideas`

`view.scope.ideas` is deprecated since we support multiple data sources instead of a single scope

adessy committed:
IN-696 Add helper source_projects association to insights views

adessy committed:
IN-696 Adapt batch_assignments_spec.rb to multi-project views

adessy committed:
IN-696 Improve batch-assigments specs

Speed up, remove some unused clutter and rubocop.

adessy committed:
IN-696 Remove references to (deprecated) view scope from specs

adessy committed:
IN-696 Adapt inputs_spec.rb to multi-project views

adessy committed:
IN-696 Remove scope methods from view

Remove scaffolding methods that were introduced to help with the transition from views with a single scope to views with multiple data sources.

adessy committed:
Clean API documentation

brentguf committed:
isValidPhoneNumber validation

brentguf committed:
isValidPhoneNumber validation

brentguf committed:
Invalid phone number error

brentguf committed:
Invalid phone number error

brentguf committed:
Add phone number validation + error to pw sign in

brentguf committed:
Add phone number validation + error to pw sign in

brentguf committed:
phoneNumberError correction

brentguf committed:
phoneNumberError correction

Commit:
Translations updated by CI (extract-intl)

Commit:
Translations updated by CI (extract-intl)

brentguf commented on front/app/components/SignUpIn/SignIn/PasswordSignin.tsx:
```diff
@@ -159,31 +162,47 @@ class PasswordSignin extends PureComponent<
+    const hasPhoneNumberError =
+      phoneLoginEnabled && isValidPhoneNumber(emailOrPhone);
+    const hasEmailError =
+      (!phoneLoginEnabled || (phoneLoginEnabled && hasPhoneNumberError)) &&
```
Wrong, needs to be changed

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Remove unused withRouter PasswordSignin

brentguf committed:
Remove unused withRouter PasswordSignin

brentguf committed:
Slight type improvements PasswordSignup

brentguf committed:
Slight type improvements PasswordSignup

brentguf committed:
PasswordSignup isPhoneSignupEnabled

brentguf committed:
PasswordSignup isPhoneSignupEnabled

brentguf committed:
PasswordSignup isPhoneSignupEnabled boolean type

brentguf committed:
PasswordSignup isPhoneSignupEnabled boolean type

brentguf committed:
getHasEmailOrPhoneNumberValidationError

brentguf committed:
getHasEmailOrPhoneNumberValidationError

brentguf committed:
Var renaming

brentguf committed:
Var renaming

brentguf committed:
Correct number of argument handleOnSubmit

brentguf committed:
Correct number of argument handleOnSubmit

brentguf committed:
Improve/clarify/correct error messaging when phone login is enabled

brentguf committed:
Improve/clarify/correct error messaging when phone login is enabled

brentguf committed:
validate argument rename

brentguf committed:
validate argument rename

brentguf committed:
emailOrPhoneNumberError state setting simplification

brentguf committed:
emailOrPhoneNumberError state setting simplification

Commit:
Translations updated by CI (extract-intl)

Commit:
Translations updated by CI (extract-intl)

brentguf committed:
Merge branch 'CL-221/error-phone-email-field-sign-up-in' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Simplify validation functions

brentguf committed:
Simplify validation functions

brentguf committed:
emailError rename to emailOrPhoneNumberError

brentguf committed:
emailError rename to emailOrPhoneNumberError

brentguf committed:
Define isPhoneSignupEnabled in 1 place

brentguf committed:
Define isPhoneSignupEnabled in 1 place

brentguf committed:
Extract functionality from handleOnSubmit

brentguf committed:
Extract functionality from handleOnSubmit

brentguf committed:
Rename email to emailOrPhoneNumber

brentguf committed:
Rename email to emailOrPhoneNumber

brentguf committed:
hasEmailOrPhoneNumberValidationError

brentguf committed:
hasEmailOrPhoneNumberValidationError

brentguf committed:
Add phone number validation + error to PasswordSignup

brentguf committed:
Add phone number validation + error to PasswordSignup

brentguf committed:
Remove wrong message

brentguf committed:
Remove wrong message

Commit:
Translations updated by CI (extract-intl)

Commit:
Translations updated by CI (extract-intl)

brentguf committed:
Lint error

brentguf committed:
Lint error

brentguf committed:
Remove TS 'as'

brentguf committed:
Remove TS 'as'

brentguf commented on front/app/components/SignUpIn/SignIn/PasswordSignin.tsx:
```diff
@@ -159,53 +155,77 @@ class PasswordSignin extends PureComponent<
+
+      if (
+        this.validate(phoneLoginEnabled, email, password) &&
+        email &&
+        password
```
Not ideal imo because after the validate call we know they're strings, but still better than using 'as string' in the signIn arguments (in case we change anything in the validate function).

brentguf commented on front/app/components/SignUpIn/SignUp/PasswordSignup.tsx:
```diff
@@ -297,30 +297,39 @@ class PasswordSignup extends PureComponent<Props & InjectedIntlProps, State> {
     }
   };
 
-  handleOnSubmit = async (event: React.FormEvent) => {
```
Gave handleOnSubmit a bit less responsibility and extracted some functionality into other functions.

brentguf committed:
Changelog

IvaKop commented on front/app/utils/validate.ts:
```diff
@@ -1,3 +1,8 @@
 }
+
+export function isValidPhoneNumber(phoneNumber: string) {
+  // current back-end implementation, see phone_service.rb
```
Ideally we would ensure these are synchronised but I can't come up with an easy way to do that off the top of my head.

brentguf replied:
True, but this is unlikely to change (often).

brentguf replied:
I'll add a note in the back-end as well to inform FE folks when validation changes.

IvaKop reviewed (approved):
Nice 👍

IvaKop commented on front/app/components/SignUpIn/SignIn/PasswordSignin.tsx:
```diff
@@ -159,53 +155,77 @@ class PasswordSignin extends PureComponent<
-    this.setState({ emailError, hasEmptyPasswordError });
-
-    if (emailError && this.emailInputElement) {
+    const hasValidPhoneNumber = emailOrPhoneNumber
```
There are much better ways to do these types of validations - ideally using `yup` or a similar tool to make our lives easier. I'm keeping it on the list of possible future improvements for when we get to the forms :)

brentguf committed:
Phone/email validation comment BE

brentguf committed:
Phone/email validation comment BE

jinjagit committed:
CL-352 Naive fix for config == nil case

nTraum committed:
[No ticket] Upgrade to ros-apartment 2.11.0

- Adds support for Ruby 3
- Adds support for Rails 7

The rest of the changelog does not look super interesting: https://github.com/rails-on-services/apartment/releases

adessy committed:
Update back/engines/commercial/insights/app/policies/insights/view_policy.rb

Co-authored-by: Philipp <ntraum@fastmail.com>

adessy committed:
IN-696 Improve clarity/readability (review comment)

jinjagit committed:
CL-352 Remove comment

adessy committed:
IN-696 Improve clarity/readability (review comment)

jinjagit committed:
CL-352 Add tests of creating tenant; not taken x taken host

jinjagit committed:
CL-352 Add test of create tenant with invalid setting

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Changelog

brentguf committed:
Changelog

brentguf committed:
Merge branch 'CL-221/error-phone-email-field-sign-up-in' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
comment phone_service: front-end file reference

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-221/error-phone-email-field-sign-up-in
